lib/project_types/script/layers/infrastructure/assemblyscript_task_runner.rb

```diff
@@ -53,12 +52,53 @@ def check_node_version!
         end
 
         def compile
-          out, status = ctx.capture2e(format(SCRIPT_SDK_BUILD, source: script_source_file, name: script_name))
+          check_compilation_dependencies!
+
+          out, status = ctx.capture2e(SCRIPT_SDK_BUILD)
           raise Domain::Errors::ServiceFailureError, out unless status.success?
         end
 
+        def check_compilation_dependencies!
+          pkg = JSON.parse(File.read('../package.json'))
+          build_script = pkg.dig('scripts', 'build')
+
+          raise Errors::UnmetCompilationDepdencyError,
+            unmet_compilation_dependency_error_message(header: "Build script not found") if build_script.nil?
+
+          unless build_script.start_with?("npx shopify-scripts")
```

Comment:
Once we have guest serialization we won't need this anymore since we won't care if developers don't bake in a `shopify_runtime_allocate` export in the final script. For now, I consider that we should keep this, also this is a good warning for older scripts that don't have a build npm script. 

On that note, right now we only have Shopify devs writing scripts in the long run this will not be the case and I think we should start thinking about making some of our tools open source, namely `@shopify/scripts-toolchain-as` so that devs have access to the changes, issues, and overall conversations. Some of this can be accessed through NPM but GitHub is definitely better IMO.

Reply:
Good idea.  I don't know how to get something approved to be open source?

Reply:
Yeah, we need to open a ticket to make the repo public, but first I think there's a setup that needs to happen, like setting up the licensing. We can probably take this offline